src/main/java/com/google/javascript/gents/CommentLinkingPass.java

```diff
@@ -180,6 +210,42 @@ private Node newFloatingCommentFromBuffer() {
       return c;
     }
 
+    /**
+     * If the parent node can survive code generation with an extra EMPTY node child, then outputs
+     * floating comments by attaching to a new EMPTY node before the current node. If the parent is
+     * sensitive about its children count/order, then try to put the comment in current node. If the
+     * current node doesn't accept comments (for example a BLOCK node will ignore any comments
+     * mapped to it), then attach the comments to the parent node. The parent node might not be the
+     * best place to hold the comment but at least we don't lose the comment and we don't break code
+     * generation.
+     *
+     * @param n
+     */
+    private void outputFloatingCommentFromBuffer(Node n) {
+      Node parent = n.getParent();
```

Comment:
it the worst case scenario we hit ROOT and  ROOT.getParent() is null, so maybe add if (n == null) return;
I am ok with dropping some comments in extreme situations instead of crashing with NPE.